### PR TITLE
Upgrade setuptools for CVE-2024-6345

### PR DESCRIPTION
**Pull request by legoktm**
## Status

Ready for review

## Description of Changes

This is low severity since PyPI would need to be compromised, but good to fix.

Remove the unused pyvirtualdisplay test dependency that's incompatible with newer setuptools.

## Testing

* [x] CI passes
* [ ] Visual review

## Deployment

Any special considerations for deployment? Affects everything.

## Checklist

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container
- [x] Linting and tests (`make -C admin test`) pass in the admin development container
- [x] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass
- [x] No AppArmor rules update needed
- [x] These changes do not require documentation
- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
  - [x] setuptools 56.0.0 to 70.3.0 - @legoktm https://github.com/freedomofpress/securedrop-builder/wiki/setuptools-56.0.0-to-70.4.0
  - [x] setuptools-scm 6.0.1 to 8.1.0 - @legoktm https://github.com/freedomofpress/securedrop-builder/wiki/setuptools_scm-6.0.1-to-8.1.0
  - [x] packaging 21.3 to 24.1 - @zenmonkeykstop https://github.com/freedomofpress/securedrop-builder/wiki/packaging-21.3-to-24.1
  - [x] tomli==2.0.1 - @zenmonkeykstop (no diff review coz new dependency, but quick code review revealed nothing amiss)
  - [x] typing-extensions==4.12.2 - @legoktm https://github.com/freedomofpress/securedrop-builder/wiki/typing_extensions-4.12.2

### CODE REVIEW

**Comment by legoktm** [comment]
I'm still waiting for CI to clear, but it's looking like we'll need diff reviews for:
* setuptools 56.0.0 to 70.3.0
* setuptools-scm 6.0.1 to 8.1.0

And for the new setuptools-scm dependencies:
* packaging==24.1
* tomli==2.0.1
* typing-extensions==4.12.2

Let me double check that these versions also work for the client so we don't need to duplicate reviews.

**Comment by legoktm** [comment]
I believe all of the diff reviews are complete now \o/

File: admin/setup.py [hunk 1]
```diff
@@ -17,4 +17,24 @@
 #
 import setuptools
 
-setuptools.setup(setup_requires=["d2to1", "pbr"], d2to1=True)
+setuptools.setup(
+    name="securedrop-admin",
```
Comment (zenmonkeykstop) [inline admin/setup.py:21]: @legoktm didn't you want to update this to generic "securedrop team" contact info at one stage, same as the changelog entries? Given this is a new file it's as good a time as any.

**Review by zenmonkeykstop — DISMISSED** [review]
Approved - but one nit re: module ownership details if you still wanna change it

File: admin/setup.py (hunk 1 above)
Comment (legoktm) [reply admin/setup.py:21]: Done.

**Review by zenmonkeykstop — APPROVED** [review]
Approving.

**Pull request merged**